### ISSUE TITLE
LaBraM's Neural Decoder Implementation produces incorrect shapes

### ISSUE THREAD

**Issue by MatthewChen37** [braindecode/braindecode#783]
When `labram(neural_tokenizer=False)`, the `_PatchEmbed` class is used, which has a couple of issues:

1. It returns a tensor of shape: `(Batch, n_patchs, patch_size, channels)`, but `forward_features` expects a 3d output
2. This also means that when the dummy tensor is passed in, `self.proj` acts on a 3-dimensional tensor, which throws a PyTorch error. 
3. Unpacking in `forward_features` leads to to little number of values to unpack (see below)
4. Regardless of neural_tokenizer, the local `temporal` variable always needs to be defined (see below), but it isn't based on issue 3.
5. `cls_tokens` is a 3-dimensional tensor but since `_PatchEmbed` returns a 4-d tensor this will throw an error for the line `x = torch.cat((cls_tokens, x), dim=1)` (see below)
```
        if self.neural_tokenizer:
            batch_size, nch, n_patch, temporal = self.patch_embed.segment_patch(x).shape
        else:
            batch_size, nch, n_patch = self.patch_embed(x).shape (this returns a 4d tensor --> unpacking error)
        
        x = self.patch_embed(x)
        # add the [CLS] token to the embedded patch tokens
        cls_tokens = self.cls_token.expand(batch_size, -1, -1)

        x = torch.cat((cls_tokens, x), dim=1) # will throw an error due to mismatch between shapes if neural_tokenizer=false!

        # Positional Embedding
        if input_chans is not None:
            pos_embed_used = self.position_embedding[:, input_chans]
        else:
            pos_embed_used = self.position_embedding

        if self.position_embedding is not None:
            pos_embed = self._adj_position_embedding(
                pos_embed_used=pos_embed_used, batch_size=batch_size
            )
            x += pos_embed

        # The time embedding is added across the channels after the [CLS] token
        if self.neural_tokenizer:
            num_ch = self.n_chans
        else:
            num_ch = n_patch
        time_embed = self._adj_temporal_embedding(
            num_ch=num_ch, batch_size=batch_size, dim_embed=temporal (this will also throw an error)
        )
        x[:, 1:, :] += time_embed

        x = self.pos_drop(x)

        for blk in self.blocks:
            x = blk(x)

        x = self.norm(x)
        if self.fc_norm is not None:
            if return_all_tokens:
                return self.fc_norm(x)
            temporal = x[:, 1:, :]
            if return_patch_tokens:
                return self.fc_norm(temporal)
            return self.fc_norm(temporal.mean(1))
        else:
            if return_all_tokens:
                return x
            elif return_patch_tokens:
                return x[:, 1:]
            return x[:, 0]
```

**Comment by bruAristimunha** [comment]
super nice to report this, can you push a PR to fix this wrong behavior please? 

The inclusion of this patch layer was to ensure that the 3D input was converted to the 4D required for the model. I understand that when you use it for the neural_tokenizer=False, it generate some errors, and was something that i didn't antecipated when i push labram here.

**Comment by bruAristimunha** [comment]
we can do some quick pr with release to solve this behavior @MatthewChen37

**Comment by bruAristimunha** [comment]
should be fine now with #792
